### ISSUE TITLE
in-page smooth scrolling problems

### ISSUE THREAD

**Issue by elegon1** [hakimio/ngx-ui-tour#85]
I made a poc with ngx-ui-tour (material) and I found out that there are issues when also using the css style:
`scroll-behavior: smooth;`

**This happens:** When the next tour step highlights a button on the very bottom of the page (so scrolling is necessary to see it) the step popover is showing too early - it shows before the scrolling has finished and therefore has lots of space between its position and the button's postion.

**Possible solution:** Maybe another option would be helpfull to delay the next popover a couple of ms like in delayAfterNavigation (maybe named **delayNextStep**)?

**Comment by hakimio** [comment]
I don't think `delayNextStep` would solve your issue since you can't know how long it will take to scroll to the element on different browsers and on different operating systems.

**Comment by elegon1** [comment]
You are right, the outcome would not be perfect in some scenarios, but in our scenario it would perfectly fit. And it is a quick and easy way to make a suitable solution in many cases.

On the other hand - your argument would also apply against delayAfterNavigation, I guess  ;)

But what really speaks for my suggestion: smooth scrolling really gives a new user on a webpage a much better orientation where to find the key elements. 
We could indeed turn off smooth scrolling while beeing in the ngx-ui-tour, but it would not be the best experience for the user.
And the need to implement a custom template just for this makes the implementaton unnecessarily cumbersome, imho.

**Comment by hakimio** [comment]
`delayAfterNavigation` was introduced to deal with issues like #74 and when the page wouldn't scroll all the way to the anchor element.
Anyway, feel free to create a PR for adding `delayBeforeStepShow` and I'll take a look at it when I have time. Remember to update documentation and all change logs.

**Comment by hakimio** [comment]
@elegon1 I have introduced `smoothScroll` option in `v10.1`. Take a look at [the release announcement](https://github.com/hakimio/ngx-ui-tour/releases/tag/v10.1.0), try out the new version and let me know if it works for you the way you wanted.
